Union "Types" don't seem to be generating

Working through a large query. It seems to choke when I need to hit a field that is a union type.  Relevant piece from schema
```
{
          "kind": "UNION",
          "name": "Asset",
          "description": null,
          "fields": null,
          "inputFields": null,
          "interfaces": null,
          "enumValues": null,
          "possibleTypes": [
            {
              "kind": "OBJECT",
              "name": "Article",
              "ofType": null
            },
            {
              "kind": "OBJECT",
              "name": "Image",
              "ofType": null
            },
            {
              "kind": "OBJECT",
              "name": "Video",
              "ofType": null
            },
            {
              "kind": "OBJECT",
              "name": "Slideshow",
              "ofType": null
            },
            {
              "kind": "OBJECT",
              "name": "InteractivePromo",
              "ofType": null
            },
            {
              "kind": "OBJECT",
              "name": "InteractiveEmbed",
              "ofType": null
            },
            {
              "kind": "OBJECT",
              "name": "Promo",
              "ofType": null
            },
            {
              "kind": "OBJECT",
              "name": "BlogPost",
              "ofType": null
            },
            {
              "kind": "OBJECT",
              "name": "ExternalUrl",
              "ofType": null
            },
            {
              "kind": "OBJECT",
              "name": "Recipe",
              "ofType": null
            },
            {
              "kind": "OBJECT",
              "name": "TimesTopic",
              "ofType": null
            }
          ]
        }
```

query (it seems to not recognize that "Article" is an "Asset", Asset Type never gets generated

```

query SimpleQuery($id:String!)
 {
     program(id:$id) {
       firstLoad {
         ...onBlock
       }
     }
   }

   fragment onBlock on Block {
     title
     showTitle
     showMore
     showSection
     firstLoad {
       ... onAssetsConnection
     }
     moreAssets {
       ... onAssetsConnection
     }
   }

   fragment onAssetsConnection on AssetsConnection {
     edges {
       node {

         ... on Article {
           promo_summary
         }
       }
     }
   }

```
Logs:
```
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:14: error: package type does not exist
import type.Asset;
           ^
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:78: error: cannot find symbol
    private @Nullable Asset node;
                      ^
  symbol:   class Asset
  location: class Edge
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:80: error: cannot find symbol
    public Edge(@Nullable Asset node) {
                          ^
  symbol:   class Asset
  location: class Edge
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:84: error: cannot find symbol
    public @Nullable Asset node() {
                     ^
  symbol:   class Asset
  location: class Edge
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:130: error: cannot find symbol
      @Nonnull Edge create(@Nullable Asset node);
                                     ^
  symbol:   class Asset
  location: interface Creator
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:121: error: cannot find symbol
        Asset node;
        ^
  symbol:   class Asset
  location: class __ContentValues
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:66: error: cannot find symbol
      public @Nonnull Edge create(@Nullable Asset node) {
                                            ^
  symbol: class Asset
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:92: error: cannot find symbol
        Field.forObject("node", "node", null, true, new Field.ObjectReader<Asset>() {
                                                                           ^
  symbol:   class Asset
  location: class Mapper
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:93: error: cannot find symbol
          @Override public Asset read(final ResponseReader reader) throws IOException {
                           ^
  symbol: class Asset
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:94: error: package Asset does not exist
            return new Asset.Mapper(factory.assetFactory()).map(reader);
                            ^
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:94: error: cannot find symbol
            return new Asset.Mapper(factory.assetFactory()).map(reader);
                                           ^
  symbol:   method assetFactory()
  location: variable factory of type Factory
/Users/Nakhimovich/apollo-android-copy/apollo-sample/build/generated/source/apollo/fragment/OnAssetsConnection.java:111: error: cannot find symbol
                contentValues.node = (Asset) value;
                                      ^
  symbol: class Asset
```

--- a/apollo-compiler/src/main/kotlin/com/apollographql/android/compiler/ir/Field.kt
+++ b/apollo-compiler/src/main/kotlin/com/apollographql/android/compiler/ir/Field.kt
@@ -42,7 +42,7 @@ data class Field(
 
   private fun toTypeName(responseType: String, typesPackage: String,
       customScalarTypeMap: Map<String, String>): TypeName {
-    val packageName = if (fields?.any() ?: false || hasFragments()) "" else typesPackage
+    val packageName = if (isNonScalar()) "" else typesPackage
     return JavaTypeResolver(customScalarTypeMap, packageName).resolve(responseType, isOptional())
   }
 
@@ -71,7 +71,7 @@ data class Field(
 
   fun isNonScalar() = hasFragments() || (fields?.any() ?: false)
 
-  fun hasFragments() = fragmentSpreads?.any() ?: false
+  fun hasFragments() = (fragmentSpreads?.any() ?: false) || (inlineFragments?.any() ?: false)
 
   fun isOptional(): Boolean = isConditional || !methodResponseType().endsWith("!")
 }

